service/k8sclient/k8sclient.go

```diff
@@ -543,6 +572,10 @@ func (c *K8sClient) UpdateXtraDBCluster(ctx context.Context, params *XtraDBParam
 		cluster.Spec.ProxySQL.Resources = c.updateComputeResources(params.ProxySQL.ComputeResources, cluster.Spec.ProxySQL.Resources)
 	}
 
+	if params.HAProxy != nil {
+		cluster.Spec.HAProxy.Resources = c.updateComputeResources(params.HAProxy.ComputeResources, cluster.Spec.HAProxy.Resources)
```

Comment:
probably we should check if `cluster.Spec.HAProxy` is exists and return error if it's not. 
Also we should check if cluster is not trying to run proxySQL and HAProxy in the same time.